GiniVision/Classes/Core/GiniScreenAPICoordinator.swift

```diff
@@ -234,24 +237,28 @@ extension GiniScreenAPICoordinator: UINavigationControllerDelegate {
                               animationControllerFor operation: UINavigationControllerOperation,
                               from fromVC: UIViewController,
                               to toVC: UIViewController) -> UIViewControllerAnimatedTransitioning? {
-        if fromVC is AnalysisViewController && operation == .pop {
+        if fromVC is AnalysisViewController {
             analysisViewController = nil
-            visionDelegate?.didCancelAnalysis()
+            if operation == .pop {
+                visionDelegate?.didCancelAnalysis()
+            }
         }
         
         if fromVC is ReviewViewController && operation == .pop {
-            // This can only happen when not using multipage
-            reviewViewController = nil
-            
+            reviewViewController = nil            
             if let firstDocument = pages.first?.document {
                 visionDelegate?.didCancelReview(for: firstDocument)
             }
         }
         
-        if toVC is CameraViewController && (fromVC is ReviewViewController || fromVC is AnalysisViewController) {
+        if toVC is CameraViewController &&
+            (fromVC is ReviewViewController ||
+             fromVC is AnalysisViewController ||
+             fromVC is ImageAnalysisNoResultsViewController) {
```

Comment:
Same remark about checking for "Single Page".

Reply:
Answered above.